tests/Integration/Behaviour/Features/Context/CatalogPriceRuleContext.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+* Copyright since 2007 PrestaShop SA and Contributors
+* PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+*
+* NOTICE OF LICENSE
+*
+* This source file is subject to the Open Software License (OSL 3.0)
+* that is bundled with this package in the file LICENSE.md.
+* It is also available through the world-wide-web at this URL:
+* https://opensource.org/licenses/OSL-3.0
+* If you did not receive a copy of the license and are unable to
+* obtain it through the world-wide-web, please send an email
+* to license@prestashop.com so we can send you a copy immediately.
+*
+* DISCLAIMER
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to https://devdocs.prestashop.com/ for more information.
+*
+* @author    PrestaShop SA and Contributors <contact@prestashop.com>
+* @copyright Since 2007 PrestaShop SA and Contributors
+* @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+*/
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context;
+
+use Behat\Gherkin\Node\TableNode;
+use SpecificPriceRule;
+
+class CatalogPriceRuleContext extends AbstractPrestaShopFeatureContext
```

Comment:
```suggestion
class CatalogPriceRuleContext extends AbstractDomainFeatureContext
```

You can extend this one which already contains `getSharedStorage` (and other behat tools that may be helpful later)

Reply:
Ah I think I get it you wanted to keep this one separated from the other Domain context since it relies on legacy code

Maybe exceptionally we can move this method into the domain context, and add a `@todo` comment explaining that we use legacy for now until a legit CQRS command is created to handle those conditions (we'll need it to finish the form migration at some point)

Reply:
> you wanted to keep this one separated from the other Domain context since it relies on legacy code

I think we have done it before, so I suggested it, isn't it the case anymore? :fearful: 
Anyway if there was no other feedbacks to fix, I would say just leave it as it is. (But I guess its not that hard to move it, considering you will still have to do some changes anyway)

Reply:
Well since i changed other things as well added this and added todo